### DISCUSSION TITLE
Manually adjust the opening/closing time for the roller shutters in Shelly 2.5

### DISCUSSION THREAD

**Discussion by vizlipuzli** [mongoose-os-apps/shelly-homekit#1551] [Q&A]
How can I manually adjust the opening and closing times for the roller shutters after calibration? During the process, the default timing doesn’t allow my motorized valve to fully open or close. I need to be able to set the time required manually to ensure complete operation. Does anyone know if there’s a way to customize these settings?

**Comment by markirb** [comment]
similar to here you can try to adjust:

https://github.com/mongoose-os-apps/shelly-homekit/issues/985#issuecomment-2439916010
move_time_ms

you seem to misuse Window Shutters for a Valve?

**Reply by vizlipuzli** [reply]
Exactly, since there’s no valve mode available in the settings. Thanks, this example looks like it could really help!

**Reply by vizlipuzli** [reply]
I set the movement time to 30 seconds using the command curl http://192.168.1.69/rpc/Config.Set -d '{"config": {"wc1": {"move_time_ms": 30000}}}', then saved the changes and rebooted the Shelly. Now the device interface shows 'movement time: 30 s,' but unfortunately, when I press the open or close button in the interface, the relay still triggers after 5 seconds as it did after calibration. Are there any other parameters I need to set? Here is the config:   "wc1": {
    "name": "Valve",
    "in_mode": 3,
    "swap_inputs": false,
    "swap_outputs": false,
    "calibrated": true,
    "idle_power_thr": 5,
    "move_power": 2.143536,
    "move_time_ms": 30000,
    "max_ramp_up_time_ms": 5000,
    "obstruction_power_coeff": 2.5,
    "obstruction_duration_ms": 3000,
    "obstruction_time_coeff": 1.5,
    "current_pos": 100
  },

**Reply by markirb** [reply]
Most likely idle_power_thr and max_ramp….  See other similar issue with big windows. cannot say without logs
